tests/e2e/parallel/main_test.go

```diff
@@ -2290,6 +2291,1146 @@ func TestScheduledSuiteUpdate(t *testing.T) {
+			},
+		},
+		Spec: compv1alpha1.TailoredProfileSpec{
+			Title:       "Test Failure Reason",
+			Description: "Test that FailureReason appears in ComplianceCheckResult",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      customRuleName,
+					Kind:      "CustomRule",
+					Rationale: "Testing failure reason propagation",
+				},
+			},
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), tp, nil)
+	if err != nil {
+		t.Fatalf("Failed to create TailoredProfile: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), tp)
+
+	// Wait for TailoredProfile to be ready
+	err = f.WaitForTailoredProfileStatus(testNamespace, tpName, compv1alpha1.TailoredProfileStateReady)
+	if err != nil {
+		t.Fatalf("TailoredProfile failed to become ready: %v", err)
+	}
+	t.Logf("TailoredProfile %s is ready", tpName)
+
+	// Create ScanSettingBinding
+	ssb := &compv1alpha1.ScanSettingBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ssbName,
+			Namespace: testNamespace,
+		},
+		Profiles: []compv1alpha1.NamedObjectReference{
+			{
+				APIGroup: "compliance.openshift.io/v1alpha1",
+				Kind:     "TailoredProfile",
+				Name:     tpName,
+			},
+		},
+		SettingsRef: &compv1alpha1.NamedObjectReference{
+			APIGroup: "compliance.openshift.io/v1alpha1",
+			Kind:     "ScanSetting",
+			Name:     "default",
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), ssb, nil)
+	if err != nil {
+		t.Fatalf("Failed to create ScanSettingBinding: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), ssb)
+
+	// Wait for scans to complete
+	// The scan should be NON-COMPLIANT because the compliance-operator deployment likely has only 1 replica
+	suiteName := ssbName
+	err = f.WaitForSuiteScansStatus(testNamespace, suiteName, compv1alpha1.PhaseDone, compv1alpha1.ResultNonCompliant)
+	if err != nil {
+		// It might be compliant if there are 3+ replicas, which is okay for this test
+		// We just need to check that the scan completed
+		err = f.WaitForSuiteScansStatus(testNamespace, suiteName, compv1alpha1.PhaseDone, compv1alpha1.ResultCompliant)
+		if err != nil {
+			t.Fatalf("Scan did not complete: %v", err)
+		}
+		t.Log("Scan completed as compliant (deployments have 3+ replicas)")
+	} else {
+		t.Log("Scan completed as non-compliant (some deployments have <3 replicas)")
+
+		// Get the ComplianceCheckResult and verify the FailureReason appears in warnings
+		checkResultName := fmt.Sprintf("%s-%s", tpName, strings.ReplaceAll(fmt.Sprintf("custom_%s", customRuleName), "_", "-"))
+		checkResult := &compv1alpha1.ComplianceCheckResult{}
+		err = f.Client.Get(context.TODO(), types.NamespacedName{
+			Name:      checkResultName,
+			Namespace: testNamespace,
+		}, checkResult)
+		if err != nil {
+			t.Fatalf("Failed to get ComplianceCheckResult: %v", err)
+		}
+
+		// Verify the check failed
+		if checkResult.Status != compv1alpha1.CheckResultFail {
+			t.Logf("Check result status is %s, not FAIL - deployments might have 3+ replicas", checkResult.Status)
```

Comment:
If we setup a dedicated namespace and deployment, we shouldn't need to log this, and instead treat it as a failed test since the rule should fail.